src/Enum.php

```diff
@@ -2,425 +2,141 @@
-            }
+        $definition = $this->findDefinition($value);
 
-            return static::make($arguments[0])->$name();
-        }
+        if ($definition === null) {
+            $enumClass = static::class;
 
-        if (static::isValidName($name) || static::isValidValue($name)) {
-            return static::make($name);
+            throw new BadMethodCallException("There's no value {$value} defined for enum {$enumClass}, consider adding it in the docblock definition.");
         }
 
-        throw new BadMethodCallException('Call to undefined method '.static::class.'::'.$name.'()');
+        $this->value = $definition->value;
+        $this->label = $definition->label;
     }
 
-    public function __toString(): string
+    public function __get($name)
     {
-        return $this->getValue();
-    }
-
-    public function getIndex(): int
-    {
-        return $this->index;
-    }
-
-    public static function getIndices(): array
-    {
-        return array_column(static::resolve(), 'index');
-    }
+        if ($name === 'label') {
+            return $this->label;
+        }
 
-    public function getValue(): string
-    {
-        return $this->value;
+        if ($name === 'value') {
+            return $this->value;
+        }
     }
 
-    public static function getValues(): array
+    public static function __callStatic(string $name, array $arguments)
     {
-        return array_column(static::resolve(), 'value');
+        return new static($name);
     }
 
-    public function getName(): string
+    public function __call($name, $arguments)
     {
-        return $this->name;
-    }
+        if (strpos($name, 'is') === 0) {
+            $other = new static(lcfirst(str_replace('is', '', $name)));
 
-    public static function getNames(): array
-    {
-        return array_keys(static::resolve());
+            return $this->equals($other);
+        }
     }
 
-    public function isAny(array $values): bool
+    public function equals(Enum ...$others): bool
     {
-        foreach ($values as $value) {
-            if ($this->isEqual($value)) {
+        foreach ($others as $other) {
+            if (
+                get_class($this) === get_class($other)
+                && $this->value === $other->value
+            ) {
                 return true;
             }
         }
 
         return false;
     }
 
-    public function isEqual($value): bool
+    protected static function values(): array
     {
-        if (is_int($value) || is_string($value)) {
-            try {
-                $value = static::make($value);
-            } catch (InvalidValueException $error) {
-                return false;
-            } catch (InvalidIndexException $error) {
-                return false;
-            }
-        }
-
-        if ($value instanceof $this) {
-            return $value->getValue() === $this->getValue();
-        }
-
-        return false;
+        return [];
     }
 
-    public function jsonSerialize(): string
```

Comment:
I think that allow to JSON serialize an enum is a pretty useful feature and we could still return `value` which would be the `int|string` stored into DB or used in requests as well.

Reply:
You're right, there's no harm in adding it.

Reply:
https://github.com/spatie/enum/pull/56/commits/b3fb70680da2a517149e9a328e32b0289faf1d39